packages/react-instantsearch-hooks/src/utils/invariant.ts

```diff
@@ -7,11 +7,15 @@ export function invariant(
   condition: boolean,
   message: string | (() => string)
 ) {
-  if (!__DEV__) {
+  if (condition) {
     return;
   }
 
-  if (!condition) {
+  if (!__DEV__) {
+    throw new Error('Invariant failed');
+  }
```

Comment:
This is fine for now, but we should improve the DX at some point so users can easily debug production errors. Granted, they can reproduce in dev to get the full error message, but if they're using error reporting services like Sentry that stacks errors and labels them by error messages, it's really convenient to be able to get the main reason right away. Also useful in difficult to reproduce situations.

I like how [React ~~does~~ dit it](https://reactjs.org/docs/error-decoder.html/) (the decoder no longer seems to work). We could have good 'ol error numbers associated with invariants and have a comprehensive list documented with the full message, explanations of what went wrong, and possible leads to solve the problem.

Reply:
it's mentioned in the description, but I indeed agree that the least we can do is have error codes, but a slightly more advanced code could collect all invariants and warnings, output them in a markdown or similar and have them accessible. Then the shortened message can include a link to that location.

In algolisearch v3 we actually had this system: 
- https://github.com/algolia/algoliasearch-client-javascript/wiki/Deprecated
- https://github.com/algolia/algoliasearch-client-javascript/blob/v3/src/deprecatedMessage.js#L6

Reply:
@Haroenv I agree, it would be nice to generate them. We could also leverage our `alg.li` to have shorter links and reduce size even more.

Reply:
Exactly, I touched this point in my PR description.

This change is just a more streamlined behavior for both development and production – we used to keep the flow going which is arguably worse because it could either end up generating cryptic errors or worse, end up in inconsistent behaviors that should have errored. This change is a preparation for better error management too.